packages/type-merging-directives/src/typeMergingDirectives.ts

```diff
@@ -0,0 +1,34 @@
+import { GraphQLSchema } from 'graphql';
+
+import { SubschemaConfig } from '@graphql-tools/delegate';
+
+import { TypeMergingDirectivesOptions } from './types';
+
+import { defaultTypeMergingDirectiveOptions } from './defaultTypeMergingDirectiveOptions';
+import { typeMergingDirectivesValidator } from './typeMergingDirectivesValidator';
+import { typeMergingDirectivesTransformer } from './typeMergingDirectivesTransformer';
+
+export function typeMergingDirectives(
+  options: TypeMergingDirectivesOptions = {}
+): {
+  typeMergingDirectivesTypeDefs: string;
+  typeMergingDirectivesValidator: (schema: GraphQLSchema) => GraphQLSchema;
+  typeMergingDirectivesTransformer: (subschemaConfig: SubschemaConfig) => SubschemaConfig;
+} {
+  const finalOptions = {
+    ...defaultTypeMergingDirectiveOptions,
+    ...options,
+  };
+
+  const { baseDirectiveName, computedDirectiveName, mergeDirectiveName } = finalOptions;
+
+  return {
+    typeMergingDirectivesTypeDefs: `
+      directive @${baseDirectiveName}(selectionSet: String!) on OBJECT
+      directive @${computedDirectiveName}(selectionSet: String!) on FIELD_DEFINITION
+      directive @${mergeDirectiveName}(argsExpr: String) on FIELD_DEFINITION
```

Comment:
I’ve got to say, I do appreciate that there are only three. That’s 25% less surface area that federation, although the merge directive takes pretty complex arguments.

Reply:
Still planning on adding your simple version with some adjustments, ie separate arguments for when you want to use simple and complex keys.... That way we don't need a separate scaler to represent a string or list of strings

Reply:
Although to be honest if you are using the directives I think it is just easier to set up resolvers that take keys and use the merge directive with no arguments. I'm not sure why you wouldn't do that

Reply:
The only use case I can see is if you want to use directives to transform a remote schema but it's difficult to add directives anyway to a remote schema.we do need to work on a general solution to that so I suppose itmakes sense to have the option to use arguments but I think it would be in advance case

Reply:
And I think the best way to set this up for remotes schema is just to use the existing configuration options on the gateway